src/segment_reshape/geometry/replace_segment.py

```diff
@@ -17,76 +17,212 @@
+    editable_ids_token = _set_editable_layer_ids.set(set_editable_ids)
+    try:
+        yield
+        for layer in layers:
+            layer.endEditCommand()
+    except GeometryTransformationError as e:
+        for layer in layers:
+            layer.destroyEditCommand()
+            if layer.id() in set_editable_ids:
+                layer.rollBack()
+        raise e
+    finally:
+        _current_edit_command_layers.reset(token)
+        _set_editable_layer_ids.reset(editable_ids_token)
 
-    # endeditcommand
 
+def _set_editable_and_begin_edit_command_once(layer: QgsVectorLayer) -> None:
+    if layer.isEditCommandActive():
+        return
 
-def _replace_segments(
-    to_replace: List[SegmentCommonFeature], new_segment: QgsLineString
+    if not layer.isEditable():
+        if not layer.startEditing():
+            raise GeometryTransformationError(f"could not start editing on {layer}")
+        _set_editable_layer_ids.get().add(layer.id())
+
+    layer.beginEditCommand("Reshape segment")
+    _current_edit_command_layers.get().append(layer)
+
+
+def make_edits_for_new_segment(
+    segment_common_parts: List[SegmentCommonPart],
+    segment_edges: List[SegmentEdge],
+    reshaped_segment: Union[QgsPoint, QgsLineString],
 ) -> None:
-    pass
+    # no crs support yet
 
-    # for result in to_replace
-    #   set editable
-    #   do edit (replace_geometry_partially())
-    #   if reversed: kutsutaan replace_geometry_partially käännetyllä new_segmentillä
+    with _wrap_all_edit_commands():
+        _replace_segments(segment_common_parts, reshaped_segment)
+        if isinstance(reshaped_segment, QgsPoint):
+            _move_edges(segment_edges, reshaped_segment, reshaped_segment)
+        else:
+            _move_edges(
+                segment_edges,
+                reshaped_segment.startPoint(),
+                reshaped_segment.endPoint(),
+            )
 
 
-def _move_geometry_segment(
+def _replace_segments(
+    to_replace_common_parts: List[SegmentCommonPart],
+    new_segment: Union[QgsPoint, QgsLineString],
+) -> None:
+    for segment_common_part in to_replace_common_parts:
+        _set_editable_and_begin_edit_command_once(segment_common_part.layer)
+        new_geometry = _replace_geometry_segment(
+            segment_common_part.feature.geometry(),
+            segment_common_part.vertex_indices,
+            (
+                new_segment.reversed()
+                if isinstance(new_segment, QgsLineString)
+                and segment_common_part.is_reversed
+                else new_segment
+            ),
+        )
+        _update_geometry_to_layer_feature(
+            segment_common_part.layer,
+            segment_common_part.feature,
+            new_geometry,
+        )
+
+
+def _replace_geometry_segment(
     original: QgsGeometry,
-    from_vertex_id: Any,
-    to_vertex_id: Any,
-    new_segment: QgsLineString,
+    vertex_indices: List[int],
+    reshaped_segment: Union[QgsPoint, QgsLineString],
```

Comment:
Related to fist comment, this function could be just splitted to two different function: _replace_geometry_segment and _replace_vertex?